### ISSUE TITLE
Play pause stop cancel functionality manual

### ISSUE THREAD

**Issue by ronaksharma8** [HangfireIO/Hangfire#866]
I want to add Play, Pause, Resume and Cancel functionality using hangfire library.

I had maintained one static list which has following columns
parentJobID, ParentJobProgressStatus = false(default), childJobId, childJobProgressStatus = false (default)

For Play functionality:-

1). I will enqueue the job and will get primaryJobId
2). Then I will call BackgroundJob.ContinueWith(primaryJobId, ()=> RemovedSuccessJobs(primaryJobId));
so that whenever primaryJobId gets completed successfully, primaryJobId will be removed from the shared list.
3). Then i will create an item with primaryJobID, primaryJobStatus = false, childJobID, childJobStatus = false and add it into list.

4). Called Thread.Sleep(3000):- just increase the program processing time.

For Cancel & Pause:-

I am removing all the jobs by using static list's primaryJobID's and childJobID's.

For Resume:- 
i will again follow at the step which i did for enqueuing jobs.

Now, my questions are

1). I am not able to use, since it show HttpContext == null inside background process. Can anyone give me suggestions why this happens and what is the solution for it.?

Please see attached file for reference.
![one](https://cloud.githubusercontent.com/assets/21169800/24943070/609a29fc-1f72-11e7-86a2-63a34bd4cdb8.png)

**Comment by burningice2866** [comment]
If you really think about it, what do you honestly imagine HttpContext would look like in background job? HttpContext represents a connection between browser and server - there is no such browser connection in a background job so what it it do then? 

This is not about bashing you're question but trying to get you to think twice about what you're asking and realizing yourself how silly it is.

And for us to help you what to do instead, we need to know what you're trying to use HttpContext for in your job so we can tell you what to do instead.

**Comment by ronaksharma8** [comment]
@burningice2866  you are right, i can't use session variables into background process :).Moreover, i had changed my implementation but didn't update in this forum.
